lib/src/controls/form/color_picker/color_picker.dart

```diff
@@ -0,0 +1,1143 @@
+
+  /// The orientation of the color picker layout
+  final Axis orientation;
+
+  /// The shape of the color spectrum
+  final ColorSpectrumShape colorSpectrumShape;
+
+  /// Whether the color preview is visible
+  final bool isColorPreviewVisible;
+
+  /// The minimum allowed hue value (0-359)
+  final int minHue;
+
+  /// The maximum allowed hue value (0-359)
+  final int maxHue;
+
+  /// The minimum allowed saturation value (0-100)
+  final int minSaturation;
+
+  /// The maximum allowed saturation value (0-100)
+  final int maxSaturation;
+
+  /// Creates a new instance of [_ColorSpectrumAndPreview].
+  const _ColorSpectrumAndPreview({
+    required this.colorState,
+    required this.colorSpectrumShape,
+    required this.onColorChanged,
+    required this.orientation,
+    required this.isColorPreviewVisible,
+    required this.minHue,
+    required this.maxHue,
+    required this.minSaturation,
+    required this.maxSaturation,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return SizedBox(
+      width: (orientation == Axis.horizontal) && !isColorPreviewVisible
+          ? ColorPickerSizes.spectrum
+          : ColorPickerSizes.maxWidth,
+      height: ColorPickerSizes.spectrum,
+      child: isColorPreviewVisible
+          ? Row(
+              children: [
+                _buildSpectrum(),
+                const SizedBox(width: ColorPickerSpacing.small),
+                _buildPreviewBox(context),
+              ],
+            )
+          : Center(child: _buildSpectrum()),
+    );
+  }
+
+  Widget _buildSpectrum() {
+    return SizedBox(
+      width: ColorPickerSizes.spectrum,
+      height: ColorPickerSizes.spectrum,
+      child: colorSpectrumShape == ColorSpectrumShape.ring
+          ? ColorRingSpectrum(
+              colorState: colorState,
+              onColorChanged: onColorChanged,
+              minHue: minHue,
+              maxHue: maxHue,
+              minSaturation: minSaturation,
+              maxSaturation: maxSaturation,
+            )
+          : ColorBoxSpectrum(
+              colorState: colorState,
+              onColorChanged: onColorChanged,
+              minHue: minHue,
+              maxHue: maxHue,
+              minSaturation: minSaturation,
+              maxSaturation: maxSaturation,
+            ),
+    );
+  }
+
+  Widget _buildPreviewBox(BuildContext context) {
+    const double width = ColorPickerSizes.preview;
+    const double height = ColorPickerSizes.spectrum;
+    const double borderRadius = 4.0;
+
+    final theme = FluentTheme.of(context);
+    final color = colorState.toColor();
+
+    return Container(
+      width: width,
+      height: height,
+      decoration: BoxDecoration(
+        border: Border.all(
+          color: theme.resources.dividerStrokeColorDefault,
+        ),
+        borderRadius: BorderRadius.circular(borderRadius),
+      ),
+      child: ClipRRect(
+        borderRadius: BorderRadius.circular(borderRadius - 1),
```

Comment:
Why?

Reply:
`BorderRadius.circular(borderRadius - 1)` was a mistake. 

Upon reviewing the WinUI 3 ColorPicker again, I noticed a few differences that prompted the updates:
- The Spectrum size in the WinUI 3 Gallery is 256 X 256, whereas in the generic XAML, it was 336 x 336.
- The Checkerboard color should adapt based on the theme. Additionally, I adjusted the square size to 4.
- The Preview layout in WinUI 3 appears to follow the order: checkerboard > color > border, so I updated the layout to align with this implementation.

Preview box: 
![image](https://github.com/user-attachments/assets/29a43ee3-743f-4004-aa64-b6c1416b92d9)